ftests/core/src/main/java/org/commonjava/indy/ftest/core/jaxrs/IndyRestMapperExceptionHandlerTest.java

```diff
@@ -0,0 +1,113 @@
+            Assert.assertNotNull(response1);
+            Assert.assertThat("Test Exception Message", is(response1.readEntity(IndyRestMapperResponse.class).getMessage()));
+            Assert.assertEquals(HttpStatus.SC_INTERNAL_SERVER_ERROR, is(response1.getStatus()));
+
+
+            IndyClientHttp indyClient2 = client.module(IndyRawHttpModule.class).getHttp();
+            Response response2 = indyClient2.get("/test/exc2", Response.class);
+            Assert.assertNotNull(response2);
+            Assert.assertThat(HttpStatus.SC_BAD_REQUEST,is(response2.getStatus()));
+
+            IndyClientHttp indyClient3 = client.module(IndyRawHttpModule.class).getHttp();
+            Response response3 = indyClient3.get("/test/exc3", Response.class);
+            Assert.assertNotNull(response3);
+            Assert.assertThat("WebApplicationException Message", is(response3.readEntity(IndyRestMapperResponse.class).getMessage()));
+            Assert.assertThat(HttpStatus.SC_INTERNAL_SERVER_ERROR, is(response3.getStatusInfo().getStatusCode()));
+
+            IndyClientHttp indyClient4 = client.module(IndyRawHttpModule.class).getHttp();
+            Response response4 = indyClient4.get("/test/exc4", Response.class);
+            Assert.assertNotNull(response4);
+            Assert.assertThat("ResteasyHttpException Message", is(response4.readEntity(IndyRestMapperResponse.class).getMessage()));
+            Assert.assertThat(HttpStatus.SC_INTERNAL_SERVER_ERROR, is(response4.getStatusInfo().getStatusCode()));
+
+
+            IndyClientHttp indyClient7 = client.module(IndyRawHttpModule.class).getHttp();
+            Response response7 = indyClient7.get("/test/exc7", Response.class);
+            Assert.assertNotNull(response7);
+            Assert.assertThat("Test IOException", is(response7.readEntity(IndyRestMapperResponse.class).getCause()));
+            Assert.assertThat(HttpStatus.SC_SERVICE_UNAVAILABLE, is(response7.getStatusInfo().getStatusCode()));
+
+
+        } catch (IndyClientException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+    @Path("/test")
+    class TestExcpetionResource {
+
+        @GET
+        @Path("/exc1")
+        public Response exc1() {
+            throw new RuntimeException("Test Exception Message");
+        }
+
+        @GET
+        @Path("/exc2")
+        public Response exc2() throws IndyWorkflowException {
+            throw new IndyWorkflowException("Test Exception Message",null);
+        }
+
+        @GET
+        @Path("/exc3")
+        public Response exc3() {
+            throw new WebApplicationException("WebApplicationException Message", StatusCodes.INTERNAL_SERVER_ERROR);
+        }
+
+        @GET
+        @Path("/exc4")
+        public Response exc4() {
+            throw new ResteasyHttpException("ResteasyHttpException Message");
+        }
+
+
+        @GET
+        @Path("/exc7")
+        public Response exc7() throws IOException {
```

Comment:
We should probably have one that throws a NullPointerException, just to make sure we get a 500 response.